src/lib/storage/chunk.cpp

```diff
@@ -12,11 +12,18 @@
 
 namespace opossum {
 
+Chunk::Chunk(size_t column_count) { _segments = std::vector<std::shared_ptr<BaseSegment>>(column_count); }
+
 void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) {
   std::lock_guard<std::mutex> lock(_add_segment_lock);
   _segments.push_back(segment);
 }
 
+void Chunk::add_segment(std::shared_ptr<BaseSegment> segment, ColumnID col_id) {
+  std::lock_guard<std::mutex> lock(_add_segment_lock);
+  _segments[col_id] = segment;
+}
```

Comment:
The way we implement this now, gives us as a problem, if there is now segment at col_id before we call the method. 
Do we need this as an overload? Otherwise I suggest that we rename this to replace_segment(_at_col_id) or something similar, and definitely add a check, to test whether t there already is a segment at that given col_id

```suggestion
void Chunk::replace_segment(std::shared_ptr<BaseSegment> segment, ColumnID col_id) {
  auto replaced_segment = this->get_segment(col_id);
  if(!replace_segment = null_ptr){
  std::lock_guard<std::mutex> lock(_add_segment_lock);
  _segments[col_id] = segment;
  } else {
  throw std::runtime_error("No existing segment to replace");
}
```